nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/ssl/SSLContextFactory.java

```diff
@@ -123,10 +142,302 @@ public void init() throws IOException {
+     */
+    public final void setAttribute(String name, String value) {
+        if (name != null && value != null) {
+            attributes.put(name, value);
+        }
+    }
+
+
+    /**
+     * @param name
+     * @return value for the name
+     */
+    public final String getAttribute(final String name) {
+        return attributes.get(name);
+    }
+
+    /**
+     * Determines the SSL cipher suites to be enabled.
+     *
+     * @param requestedCiphers Comma-separated list of requested ciphers
+     * @param supportedCiphers Array of supported ciphers
+     *
+     * @return Array of SSL cipher suites to be enabled, or null if none of the requested ciphers are supported
+     */
+    protected String[] getEnabledCiphers(String requestedCiphers, String[] supportedCiphers) {
+        String[] enabled = null;
+        if (requestedCiphers != null) {
+            List<String> vec = null;
+            String cipher = requestedCiphers;
+            int index = requestedCiphers.indexOf(',');
+            if (index != -1) {
+                int fromIndex = 0;
+                while (index != -1) {
+                    cipher = requestedCiphers.substring(fromIndex, index).trim();
+                    if (cipher.length() > 0) {
+                        /*
+                         * Check to see if the requested cipher is among the
+                         * supported ciphers, i.e., may be enabled
+                         */
+                        for (int i = 0; supportedCiphers != null && i < supportedCiphers.length; i++) {
+                            if (supportedCiphers[i].equals(cipher)) {
+                                if (vec == null) {
+                                    vec = new ArrayList<>();
+                                }
+                                vec.add(cipher);
+                                break;
+                            }
+                        }
+                    }
+                    fromIndex = index + 1;
+                    index = requestedCiphers.indexOf(',', fromIndex);
+                } // while
+                cipher = requestedCiphers.substring(fromIndex);
+            }
+
+            assert cipher != null;
+
+            cipher = cipher.trim();
+            if (cipher.length() > 0) {
+                /*
+                 * Check to see if the requested cipher is among the
+                 * supported ciphers, i.e., may be enabled
+                 */
+                for (int i = 0; supportedCiphers != null
+                    && i < supportedCiphers.length; i++) {
+                    if (supportedCiphers[i].equals(cipher)) {
+                        if (vec == null) {
+                            vec = new ArrayList<>();
+                        }
+                        vec.add(cipher);
+                        break;
+                    }
+                }
+            }
+            if (vec != null) {
+                enabled = vec.toArray(new String[vec.size()]);
```

Comment:
Maybe better use `vec.toArray(String[]::new)`?

Reply:
I tried to avoid touching too much to avoid mistakes, but yes, usually I do these changes. I will do that.

Reply:
I did that in tho whole module now.